python/nwis_client/hydrotools/nwis_client/iv.py

```diff
@@ -12,58 +12,113 @@
 
 import datetime
 from collections.abc import Iterable
-from multiprocessing import Pool
 from functools import partial
-from typing import List, Tuple, Union, Iterable
+from typing import Dict, List, Set, T, Union, Iterable
 import re
 import requests
 import six
 import warnings
 
 import numpy as np
 import pandas as pd
-from hydrotools._restclient import RestClient
-from .constants import US_STATE_AND_TERRITORY_ABBREVIATIONS
+from hydrotools._restclient import RestClient, Url
+from collections.abc import Sequence
 
 
 class IVDataService:
-    """A REST client class.
-    The IVDataService class provides various methods for constructing
-    requests, retrieving data, and parsing responses from the NWIS IV
-    Service.
+    """
+    Provides a programatic way to retrieve USGS Instantanous Value Service data in the
+    canonical hydrotools pandas dataframe format. The IVDataService implements an
+    sqlite3 request cache and asynchronous request backend that splits up and retrieves
+    requests concurrently.
 
     Parameters
     ----------
-    processes : int
-        Max multiprocessing processes, default 3
-    retry : int
-        Max number of, per request retries, default 3
+    enable_cache : bool
+        Toggle sqlite3 request caching
+    cache_expire_after : int
+        Cached item life length in seconds
+
+    Examples
+    --------
+    >>> from hydrotools.nwis_client import IVDataService
+    >>> service = IVDataService()
```

Comment:
Note: The `README.md` example will need to be updated since it currently calls `IVDataService.get` directly as a class method.

Reply:
I think I already got everything with fc552637938fb997d8ad9703c426aecd46df6da9, am I missing something else?